model_analyzer/result/result_comparator.py

```diff
@@ -146,33 +138,30 @@ def compare_measurements(self, measurement1, measurement2):
         non_gpu_data1 = measurement1.non_gpu_data()
         non_gpu_data2 = measurement2.non_gpu_data()
 
-        for priority in self._metric_priorities:
-            if priority in self._non_gpu_type_to_idx:
-                # Get position in measurements of current priority's value
-                metric_idx = self._non_gpu_type_to_idx[priority]
-                threshold = self._comparison_threshold_factor * non_gpu_data1[
-                    metric_idx].value()
+        score_diff = 0.0
+        for objective, weight in self._metric_objectives.items():
+            if objective in self._non_gpu_type_to_idx:
+                metric_idx = self._non_gpu_type_to_idx[objective]
                 value_diff = (non_gpu_data1[metric_idx] -
                               non_gpu_data2[metric_idx]).value()
-
-                if value_diff > threshold:
-                    return 1
-                elif value_diff < -threshold:
-                    return -1
-            elif priority in self._gpu_type_to_idx:
-                metric_idx = self._gpu_type_to_idx[priority]
-                threshold = self._comparison_threshold_factor * avg_gpu_data1[
-                    metric_idx].value()
+                value_sum = (non_gpu_data1[metric_idx] +
```

Comment:
You don't add the two values for the denominator. The equation to compare run A to run B will be...

```
Score = sum^0_n{w_i * (B_i - A_i) / (A_i)}
```
Where `w_i` i s the weight of the metric, `B_i` is the metric value for run B, and `A_i` is the is the metric value for run A.

The equation `(B_i - A_i) / (A_i)` is commonly used it research to determine the gain one test over another. For example if A_i = 100, and B_i = 150, then the gain is `(150-100)/100 = .5`. So in this example, B is 50% better than A. It works the other way too. A_i = 100, and B_i = 50, then the gain is `(50-100)/100 = -.5`, so B is 50% worse than A.

Reply:
Do keep in mind, that some metrics are opposite. For example, GPU memory usage, lower is better. In this case the equation becomes.
```
Score = sum^0_n{w_i * (-1)*(B_i - A_i) / (A_i)}
```